### PR TITLE
netbsd.libc: Create from constituent pkgs not hack

### PR DESCRIPTION
**Pull request by Ericson2314**
## Description of changes

The old `postInstall` was ugly, and made for a needlessly course-grained package. This makes more smaller derivations, and ones that additionally can be built in parallel.

## Things done

Cross compiled Nix 2.19

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/os-specific/bsd/netbsd/pkgs/libc.nix [hunk 1]
```diff
@@ -1,102 +1,52 @@
...
-
-    make -C $BSDSRCDIR/lib/libutil $makeFlags
-    make -C $BSDSRCDIR/lib/libutil $makeFlags install
-
-    make -C $BSDSRCDIR/lib/librt $makeFlags
-    make -C $BSDSRCDIR/lib/librt $makeFlags install
+  outputs = [
+    "out"
+    "dev"
+    "man"
```
Comment (rhelmot) [inline pkgs/os-specific/bsd/netbsd/pkgs/libc.nix:24]: Is symlinkJoin smart enough to actually use all three of these outputs?

Comment (Ericson2314) [reply pkgs/os-specific/bsd/netbsd/pkgs/libc.nix:24]: Yes!

File: pkgs/os-specific/bsd/netbsd/pkgs/mkDerivation.nix [hunk 2]
```diff
@@ -28,6 +29,8 @@ lib.makeOverridable (
         stdenvNoCC
       else if attrs.noLibc or false then
         stdenvNoLibc
+      else if attrs.libc-minimal or false then
```
[the remark this answers is not in the document]
Comment (Ericson2314) [reply pkgs/os-specific/bsd/netbsd/pkgs/mkDerivation.nix:32]: Oh you like the camel case better!

[the remark this answers is not in the document]
Comment (rhelmot) [reply pkgs/os-specific/bsd/netbsd/pkgs/mkDerivation.nix:32]: If kebab-case is preferred by convention we should probably switch noLibc and noLibcxx to that as well.

File: pkgs/os-specific/bsd/netbsd/default.nix [hunk 3]
```diff
@@ -140,25 +149,12 @@ makeScopeWithSplicing' {
...
-        "lib/libm"
-        "lib/libpthread"
-        "lib/libresolv"
-        "lib/librpcsvc"
-        "lib/librt"
-        "lib/libutil"
-        "libexec/ld.elf_so"
-        "sys"
-      ];
+      #libc = abort "asdf" self.libc-minimal;
```
Comment (rhelmot) [inline pkgs/os-specific/bsd/netbsd/default.nix:154]: dead debug code

File: pkgs/os-specific/bsd/netbsd/pkgs/mkDerivation.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (Ericson2314) [reply pkgs/os-specific/bsd/netbsd/pkgs/mkDerivation.nix:32]: I see `systemdMinimal` so there is some precedent.

I think there is sort of a soft "kebab for pkg names, camel for misc things" convention.

**Pull request merged**